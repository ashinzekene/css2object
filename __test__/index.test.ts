import Css2Object from '../build';
import fs from 'fs';
import path from 'path';

describe('CSS2Object', () => {
	const cssData = fs.readFileSync(path.resolve('__test__', 'codemirror.css'), {
		encoding: 'utf-8'
	});

	it('renders correctly', () => {
		const css2json = new Css2Object(cssData);
		css2json.read();
		expect(css2json.cssRecord).toMatchSnapshot();
	});

	it('recognizes tags without a class or id', () => {
		const css2json = new Css2Object(cssData);
		css2json.read();
		expect(css2json.cssRecord['body']).toEqual({
			color: '#000',
			'font-family': 'tahoma, arial, sans-serif',
			margin: '0',
			padding: '3em 6em'
		});
	});

	it('recognizes tags whose parent has an id', () => {
		const css2json = new Css2Object(cssData);
		css2json.read();
		expect(css2json.cssRecord['#navigation a']).toEqual({
			'font-weight': 'bold',
			'text-decoration': 'none !important'
		});
	});

	it('could call a static method to return a JSON string', () => {
		const json = Css2Object.toJSON(cssData);
		const parsedJson = JSON.parse(json);
		expect(parsedJson).toHaveProperty('body');
		expect(parsedJson).toHaveProperty('h1:before');
		expect(parsedJson).toHaveProperty('h2');
	});
});
