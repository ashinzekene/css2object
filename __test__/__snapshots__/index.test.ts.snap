// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CSS2Object renders correctly 1`] = `
Object {
  "#navigation a": Object {
    "font-weight": "bold",
    "text-decoration": "none !important",
  },
  "body": Object {
    "color": "#000",
    "font-family": "tahoma, arial, sans-serif",
    "margin": "0",
    "padding": "3em 6em",
  },
  "code": Object {
    "color": "#418a8a",
    "font-family": "courier, monospace",
    "font-size": "80%",
  },
  "h1": Object {
    "font-size": "2.5em",
  },
  "h1:before": Object {
    "content": "'",
  },
  "h2": Object {
    "font-size": "1.7em",
  },
  "h2:before": Object {
    "content": "'",
  },
}
`;
